app/controllers/api/v2/registration_commands_controller.rb

```diff
@@ -15,7 +15,7 @@ class RegistrationCommandsController < V2::BaseController
         param :smart_proxy_id, :number, desc: N_("ID of the Smart Proxy. This Proxy must have enabled both the 'Templates' and 'Registration' features")
         param :setup_insights, :bool, desc: N_("Set 'host_registration_insights' parameter for the host. If it is set to true, insights client will be installed and registered on Red Hat family operating systems")
         param :setup_remote_execution, :bool, desc: N_("Set 'host_registration_remote_execution' parameter for the host. If it is set to true, SSH keys will be installed on the host")
-        param :jwt_expiration, :number, desc: N_("Expiration of the authorization token (in hours)")
+        param :jwt_expiration, String, desc: N_("Expiration of the authorization token (in hours)")
```

Comment:
This should allow both: number and string as well. But I think we should go only with `:number` only, with following logic:
* `x < 0` raise an error, invalid value
* `x = 0` unlimited expiration
* `x > 0` expiration in hours.

This should be also described in the documentation

Reply:
Also forget, if not set at all, default value should be 4 (same as in the UI)

Reply:
That makes sense @stejskalleos , thank you

Reply:
What about leaving negative numbers as valid, but treat them as `unlimited expiration`? I mean, `0` can be used for test purposes to simulate expired tokens without a need to actually wait.

Reply:
@ofedoren , currently 0 is not accepted in the UI , do we update that as well?

Reply:
The Api documentation (ss attached)  accepts jwt_expiration as a string. Currently the value of jwt_args[:expiration] is a nil when the value is unlimited, can we replace nil with '-1' without any change in the output being generated? 
![Screenshot from 2023-12-13 13-06-58](https://github.com/theforeman/foreman/assets/115526987/3b90b511-a18d-4e64-a546-5ae49db73da6)

Reply:
To note: The value of `jwt_args[:expiration]` (final converted value for jwt_expiration) for -1 as per this  [line](https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/foreman/controller/registration_commands.rb#L36) would be -3600. Would that be customary to add for unlimited hours?

Reply:
As discussed with @ares on our 1:1, it's a good idea to verify with the QEs if they  actually use 0 for testing purposes (from UI) , can double check for api as well, accordingly we can either use '-1' or '0'. I will also verify the convention with the reporter and update the PR.